pkgs/applications/graphics/loupe/default.nix

```diff
@@ -0,0 +1,94 @@
+{ stdenv
+, lib
+, fetchFromGitLab
+, rustPlatform
+, meson
+, pkg-config
+, gtk4
+, itstool
+, desktop-file-utils
+, ninja
+, wrapGAppsHook4
+, libadwaita
+, gnome
+, webp-pixbuf-loader
+, gdk-pixbuf
+, libgweather
+, librsvg
+, shared-mime-info
+, libheif
+, libxml2
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "loupe";
+  version = "unstable-2023-03-14";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "Incubator";
+    repo = "loupe";
+    rev = "889c818c14179e3f4755fb56f38d1af603e009cf";
+    hash = "sha256-9FNUAypm2GAd8/deEbNhCV+eIEGxOBCzFyytKaK9dnw=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) src;
+    hash = "sha256-tUi+y6raAhfG1bvUuSR/MRIax3OhExOBxn8gElLBRfI=";
+  };
```

Comment:
```suggestion
  cargoDeps = rustPlatform.importCargoLock {
    lockFile = finalAttrs.src + /Cargo.lock;
    outputHashes = {
      "librsvg-2.56.0" = "sha256-4poP7xsoylmnKaUWuJ0tnlgEMpw9iJrM3dvt4IaFi7w=";
    };
  };
```

Reply:
> `lockFile = finalAttrs.src + /Cargo.lock;`

This will use import-from-derivation, which is not allowed in Nixpkgs.

Reply:
Yeah need to vendor the cargo.lock in this case